src/415D90.c

```diff
@@ -3534,60 +3610,864 @@ void btl_state_draw_player_menu(void) {
+            gBattleSubState = BATTLE_SUB_STATE_PARTNER_UNUSED_CHANGE_PARTNER_4;
+        } else if (func_802A11B0() != 0) {
+            gBattleSubState = battleStatus->unk_6E;
+            btl_state_update_partner_menu();
+            btl_set_state(BATTLE_STATE_CHANGE_PARTNER);
+        }
+        break;
+    case BATTLE_SUB_STATE_PARTNER_UNUSED_CHANGE_PARTNER_4:
+        if (func_802A11B0() != 0) {
+            gBattleSubState = BATTLE_SUB_STATE_PARTNER_UNUSED_CHANGE_PARTNER_2;
+        }
+        break;
+    case BATTLE_SUB_STATE_PARTNER_UNUSED_CHANGE_PARTNER_5:
+        func_800F16CC();
+        func_802A1098();
+        gBattleSubState = BATTLE_SUB_STATE_PARTNER_UNUSED_CHANGE_PARTNER_4;
+        break;
+    case BATTLE_SUB_STATE_PARTNER_UNUSED_CHANGE_PARTNER_6:
+        destroy_popup_menu();
+        func_802A10B8();
+        break;
+    case BATTLE_SUB_STATE_PARTNER_UNUSED_CHANGE_PARTNER_7:
+        if (popupMenu->result != -1) {
+            gBattleSubState = BATTLE_SUB_STATE_PARTNER_UNUSED_CHANGE_PARTNER_2;
+        }
+        break;
+    case BATTLE_SUB_STATE_PARTNER_FOCUS_1:
+        numOptions = 1;
+        battleStatus->submenuMoves[0] = MOVE_FOCUS;
+        battleStatus->submenuMoveCount = numOptions;
+        battleStatus->submenuIcons[0] = ITEM_PARTNER_ATTACK;
+        battleStatus->submenuStatus[0] = 1;
+        for (i = 0; i < battleStatus->submenuMoveCount; i++) {
+            moveData = &gMoveTable[battleStatus->submenuMoves[i]];
+            battle_menu_moveOptionIconScripts[i] = battle_menu_StarPowerMovesHudScripts[(battleStatus->submenuMoves[i] - MOVE_FOCUS) * 2 + 0];
+            if (battleStatus->submenuStatus[i] == 0) {
+                battle_menu_moveOptionIconScripts[i] = battle_menu_StarPowerMovesHudScripts[(battleStatus->submenuMoves[i] - MOVE_FOCUS) * 2 + 1];
+            }
+            battle_menu_moveOptionDisplayCosts[i] = moveData->costFP;
+            battle_menu_moveOptionBPCosts[i] = moveData->costBP;
+            battle_menu_moveIndices[i] = i;
+            battle_menu_moveOptionSortPriorities[i] = i;
+            battle_menu_moveOptionsEnabled[i] = battleStatus->submenuStatus[i];
+            battle_menu_moveOptionNames[i] = moveData->nameMsg;
+            battle_menu_moveOptionDescriptions[i] = moveData->shortDescMsg;
+            battle_menu_moveOptionDisplayCostReductions[i] = 0;
+            battle_menu_moveOptionDisplayCostReductionColors[i] = 0;
+        }
+        if (battleStatus->unk_5C[6] < 0) {
+            battleStatus->unk_5C[6] = 0;
+        }
+        battle_menu_hasSpiritsMenu = 1;
+        battle_menu_moveOptionCount = battleStatus->submenuMoveCount;
+        s1 = battleStatus->unk_5C[6];
+        D_802AD4A8 = s1;
+        func_802A2684();
+        gBattleSubState = BATTLE_SUB_STATE_PARTNER_FOCUS_2;
+        break;
+    case BATTLE_SUB_STATE_PARTNER_FOCUS_2:
+        numOptions = func_802A2C84();
+        if (numOptions != 0) {
+            if (numOptions == 255) {
```

Comment:
Interesting this isn't -1. Does changing `numOptions` to 8-bit allow -1 here and not break elsewhere?

Reply:
-1 and 255 are different possible results that can be returned by this kind of function. 255 means return back from submenu and -1 is for error.